crates/matrix-sdk/src/widget/machine/tests/api_versions.rs

```diff
@@ -21,18 +21,19 @@ use crate::widget::machine::{Action, IncomingMessage, WidgetMachine};
 
 #[test]
 fn get_supported_api_versions() {
-    let (mut machine, mut actions_recv) =
+    let (mut machine, _) =
         WidgetMachine::new(WIDGET_ID.to_owned(), owned_room_id!("!a98sd12bjh:example.org"), true);
 
-    machine.process(IncomingMessage::WidgetMessage(json_string!({
+    let mut actions = machine.process(IncomingMessage::WidgetMessage(json_string!({
         "api": "fromWidget",
         "widgetId": WIDGET_ID,
         "requestId": "S2ixNhjaC0kd0jJn",
         "action": "supported_api_versions",
         "data": {},
     })));
 
-    let action = actions_recv.try_recv().unwrap();
+    let action = actions.remove(0);
+    assert!(actions.is_empty());
```

Comment:
```suggestion
    let [action]: [Action; 1] = actions.try_into().unwrap();
```

Reply:
Oh, this looks much more elegant, thanks! Updated in all places.